rfcs/DisableErrorPropagationDirective.md

```diff
@@ -0,0 +1,243 @@
+
+3. **Fine-grained Error Control**: Clients may want to specify different error handling behaviors for different parts of their queries based on their application requirements.
+
+## ✅ RFC Goals
+
+- Provide a way for clients to disable error propagation for specific operations
+- Help uncouple nullability and error handling in GraphQL
+- Support the transition to more semantically accurate nullability in schemas
+- Maintain backward compatibility with existing GraphQL behavior
+- Keep the implementation simple and focused
+
+## 🚫 RFC Non-goals
+
+- This is not intended to be a general-purpose error handling solution
+
+## 🧑‍💻 Proposed Solution
+
+Add a new directive `@disableErrorPropagation` that can be applied to operations in executable documents:
+
+```graphql
+directive @disableErrorPropagation on QUERY | MUTATION | SUBSCRIPTION
+
+query GetUserPosts @disableErrorPropagation {
+  user {
+    id
+    name
+    posts {
+      id
+      title
+      content
+    }
+  }
+}
+```
+
+When this directive is present on an operation:
+1. Errors thrown during execution will still be added to the `errors` array
+2. The errors will not cause nullability violations to propagate up through parent fields
+
+### Example
+
+Given these types:
+
+```graphql
+type User {
+  id: ID!
+  name: String
+  posts: [Post!]!
+}
+
+type Post {
+  id: ID!
+  title: String!
+  content: String
+}
+```
+
+And this query:
+
+```graphql
+query GetUserPosts @disableErrorPropagation {
+  user {
+    id
+    name
+    posts {
+      id
+      title
+      content
+    }
+  }
+}
+```
+
+If the `title` resolver for one of the posts throws an error:
+
+**Current behavior** (without directive):
+```json
+{
+  "data": null,
+  "errors": [
+    {
+      "message": "Failed to load title",
+      "path": ["user", "posts", 0, "title"]
+    }
+  ]
+}
+```
+
+**With @disableErrorPropagation**:
+```json
+{
+  "data": {
+    "user": {
+      "id": "123",
+      "name": "Alice",
+      "posts": [
+        {
+          "id": "post1",
+          "title": null,
+          "content": "Some content"
+        }
+      ]
+    }
+  },
+  "errors": [
+    {
+      "message": "Failed to load title",
+      "path": ["user", "posts", 0, "title"]
+    }
+  ]
+}
+```
+
+## 🛠️ Implementation Considerations
+
+The implementation requires changes to the execution algorithm in graphql-js:
+
+1. During operation execution, check if the operation has the `@disableErrorPropagation` directive
+2. If present, modify error handling to prevent propagation while still collecting errors
+3. Return field values as-is, even if errors occurred
+
+While this proposal focuses on a simple boolean directive, future extensions might consider additional error behaviors:
+
+- `PROPAGATE`: Current default behavior (errors propagate up)
+- `NULL`: Replace errored positions with null
+- `ABORT`: Abort the entire request on any error
+- `INLINE`: Replace errored positions with an error representation
```

Comment:
`INLINE` has been removed from the proposal see [this comment](https://github.com/graphql/graphql-spec/pull/1050#issuecomment-2676191239) on the original RFC

```suggestion
```